grails-app/domain/au/org/emii/portal/Server.groovy

```diff
@@ -223,4 +222,10 @@ class Server {
             operations << operation
         }
     }
+
+    def getUrlListDownloadSubstitutions() {
+        return grailsApplication?.config?.knownServers?.find {
+            this.uri.startsWith(it.uri)
+        }?.urlListDownloadSubstitutions ?: [:]
```

Comment:
All those save navigation operators could be a smell (I would say so).

If you wanted to change this we could update the tests so that Server always has a (mocked) `grailsApplication` object. I think there are 3 tests that break, currently.

Alternatively, the following code is equivalent (assuming no-one removes the knownServers entry from their config) and doesn't use all those operators:
``` groovy
return grailsApplication ? grailsApplication.config.knownServers.find {
    this.uri.startsWith(it.uri)
}.urlListDownloadSubstitutions : [:]
```
No tests would need to be updated to make this change.